tests/suites/test_suite_aria.function

```diff
@@ -227,9 +225,9 @@ void aria_encrypt_ecb( data_t *key_str, data_t *src_str,
             TEST_ASSERT( mbedtls_aria_crypt_ecb( &ctx, src_str->x + i,
                                                  output + i ) == 0 );
         }
-        mbedtls_test_hexify( dst_str, output, src_str->len );
 
-        TEST_ASSERT( strcasecmp( (char *) dst_str, hex_dst_string ) == 0 );
+        TEST_ASSERT( !memcmp( output,
```

Comment:
Minor: I know both are equivalent, but I find `memcmp(...) == 0` (or `0 == memcmp(...)`) more readable, and it's the version that's the most widely used in our code base so far, so it's probably better to be consistent. (I won't be flagging every occurrence but there are others.)

Reply:
Actually, in new code, please use `ASSERT_COMPARE`. I'm pushing for the use of this macro after seeing too many places where we compared the data, but forgot to check for equal lengths.

I thought `!memcmp` was explicitly forbidden in our [coding standards](https://tls.mbed.org/kb/development/mbedtls-coding-standards), but now that I try to look it up I can't find it. It seems to be implied under “Functions whose name clearly indicates a boolean”, which mentions “`==` 0 and `!=` 0 tests”, but it isn't stated explicitly. In any case, it's a de facto standard that we don't use integers as booleans implicitly.

Reply:
Yep I agree with all of this. I've replaced all occurrences of `"TEST_ASSERT(!memcmp("` by `ASSERT_COMPARE`.